app/controllers/settings/external_cert_controller.rb

```diff
@@ -355,10 +357,52 @@ def warning_subjectaltname(cert, warning_messages, required_san_array, service_n
   end
   # rubocop:enable Metrics/CyclomaticComplexity, Metrics/PerceivedComplexity
 
-  # # Placeholder for trust chain validation
-  # def trust_chain_verify(_cert)
-  #   true
-  # end
+  def trust_chain_verify(_cert)
+    return _trust_chain_verify(_cert,_build_system_wide_certs_store())
+  end
+
+  def _build_system_wide_certs_store
+    cert_store = OpenSSL::X509::Store.new
+
+    system_certs = SystemCertificate.all
+    for system_cert in system_certs
+      cert = system_cert.certificate
+      cert_text = cert.certificate#get_certificate_text
+      cert_obj = read_cert(cert_text)
+      next unless cert_obj
```

Comment:
Isn't this an error condition? Shouldn't it throw an exception?

Reply:
Do we want it to error or just warn the user?

Reply:
Added a comment in code mentioning this is okay as a cert cannot get into the SystemCertificate table without having already passed a read_cert check.